aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -265,7 +265,13 @@ impl AptosVM {
             .finish(&mut (), gas_meter.change_set_configs())
             .map_err(|e| e.into_vm_status())?;
         // Charge gas for write set
-        gas_meter.charge_write_set_gas(user_txn_change_set_ext.write_set().iter())?;
+        gas_meter.charge_write_set_gas_for_io(user_txn_change_set_ext.write_set().iter())?;
+        gas_meter.charge_storage_fee(
```

Comment:
This needs to be gated

Reply:
It's gated inside `charge_write_set_gas_for_io` and `charge_storage_fee`.